metaflow/task.py

```diff
@@ -493,6 +504,19 @@ def run_step(
                     )
                 )
 
+            # Add runtime dag information to the metadata of the task
+            foreach_execution_path = self._dynamic_runtime_metadata(foreach_stack)
```

Comment:
[this](https://github.com/Netflix/metaflow/pull/2124/files#r1941819183) comment still applies